### PR TITLE
Unpinning the "packaging" library with specific python versions in requirements.txt

### LINKED ISSUE #34806: [Datasets] Import issue with `packaging` module in ray.data.extensions
### What happened + What you expected to happen

On a clean install of `pip install ray` in Python versions < 3.10

When importing `ray.data.extensions`, this fails with:

```
>>> import ray.data.extensions
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jaychia/code/Daft/venv/lib/python3.8/site-packages/ray/data/extensions/__init__.py", line 1, in <module>
    from ray.data.extensions.tensor_extension import (
  File "/Users/jaychia/code/Daft/venv/lib/python3.8/site-packages/ray/data/extensions/tensor_extension.py", line 1, in <module>
    from ray.air.util.tensor_extensions.pandas import (  # noqa: F401
  File "/Users/jaychia/code/Daft/venv/lib/python3.8/site-packages/ray/air/util/tensor_extensions/pandas.py", line 35, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'
```

### Versions / Dependencies

Python 3.8 (or any version of Python < 3.10)
Ray 2.4.0

### Reproduction script

```
import ray.data.extensions
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

Some context: https://github.com/ray-project/ray/pull/28315

The packaging library is pinned to python>=3.10, which may cause installation errors in lower versions of Python. 

Currently, there are two ways of handling versions (distutils and packaging) in our code. `distutils` is getting deprecated in 3.12 and we should in favor of `packaging.version`.

This PR unpins the packaging library in requirements and replaces the usage of `distutils.version` with `packaging.version`.

## Related issue number

Closes #34806 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/requirements.txt [hunk 1]
```diff
@@ -19,6 +19,7 @@ aiosignal
 frozenlist
 requests
 virtualenv>=20.0.24, < 20.21.1
+packaging
```
Comment (krfricke) [inline python/requirements.txt:22]: If we unpin here, we should also unpin in `setup.py`

**Review by krfricke — CHANGES_REQUESTED** [review]
Should we also remove imports of `from distutils.version import LooseVersion as Version`?

File: python/requirements.txt (hunk 1 above)
Comment (woshiyyya) [reply python/requirements.txt:22]: Oh I see. I'll unpin it and also remove all the `distutils` usage. accordingly

**Review by krfricke — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]
Yunxuan said he tested this across different python versions

**Review by bveeramani — APPROVED** [review]

**Review by ericl — APPROVED** [review]

**Pull request merged**